README.md

```diff
@@ -4,22 +4,116 @@
+* **Handles multiple languages**: No need to learn and maintain multiple tools for your polyglot environment (e.g. ESLint, find-sec-bugs, rubocop, gosec, ...).
+* **Easy to tweak**: Rules look like the code you’re searching, no static analysis PhD required.
+* Does **not** require compilation, Semgrep runs on source code directly. 
+* **Easy to integrate**: Highly portable and many CI and git hook integrations already exist. Run via CLI or Docker, output `--json` and pipe results into whatever system you want.
+
+
+### Language support
+
+| **Python** | **Javascript** | **Go &nbsp; &nbsp; &nbsp;** | **Java &nbsp;** | **C &nbsp; &nbsp; &nbsp; &nbsp;** | **Ruby** | **Typescript** | **PHP &nbsp; &nbsp;** |
+|:-----------|:---------------|:----------------------------|:----------------|:----------------------------------|:---------|:---------------|:----------------------|
+| ✅          | ✅              | ✅                           | ✅               | ✅                                 | 🚧 |  Coming...      | Coming...             |
+
+... and many more coming.
+
+### Pattern Syntax Teaser
+
+One of the most unique and useful things about Semgrep is how easy it is to write and iterate on queries. 
 
-| **Python** | **Javascript** | **Go &nbsp; &nbsp; &nbsp;** | **Java &nbsp;** | **C &nbsp; &nbsp; &nbsp; &nbsp;** | **Typescript** | **PHP &nbsp; &nbsp;** |
-|:-----------|:---------------|:----------------------------|:----------------|:----------------------------------|:---------------|:----------------------|
-| ✅          | ✅              | ✅                           | ✅               | ✅                                 | Coming...      | Coming...             |
+The goal is to make it as *easy as possible* to go from an idea in your head to finding the code patterns you intend to.
 
-Example patterns:
+**Example**: Let's say you want to find all calls to a function named `exec`, and you don't care about the arguments. With Semgrep, you could simply supply the pattern `exec(...)` and you'd match:
+
+~~~python
+# Simple cases grep could find
+exec("ls")
+exec(some_var)
+
+# But you don't have to worry about whitespace
+exec (foo)
+
+# Or calls across multiple lines
+exec (
+    bar
+)
+~~~
+
+Importantly, Semgrep would *not* match the following:
+
+~~~python
+# grep would match this, but semgrep ignores it because 
+# it doesn't have the right function name
+other_exec(bar)
+
+# semgrep ignores commented out lines
+# exec(foo)
+
+# and hard-coded strings
+print("exec(bar)")
+~~~
+
+Semgrep can even do fancy things like matching aliased imports:
+
+~~~python
+# Semgrep knows that safe_function refers to exec so it
+# will still match!
+#   Oof, try finding this with grep
+import exec as safe_function
+safe_function(tricksy)
+~~~
+
+Play with this example in your browser here: https://semgrep.live/QrkD.
```

Comment:
NICE!

Reply:
I'd love to include a "run this locally" snippet for people: `semgrep -l javascript -e "exec(...)" code`